cypress/plugins/index.ts

```diff
@@ -0,0 +1,40 @@
+// ***********************************************************
+// This example plugins/index.js can be used to load plugins
+//
+// You can change the location of this file or turn off loading
+// the plugins file with the 'pluginsFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/plugins-guide
+// ***********************************************************
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore: Could not find a declaration file for module
+import coverageTask from '@cypress/code-coverage/task';
+
+/**
+ * Performs common plugin and configuration setup - registers coverage plugin and sets `baseUrl` property if MANAGER_IP
+ * environment variable is set
+ *
+ * @param on Adds listener function to the end of the listeners array for the given event
+ *           @see https://nodejs.org/api/events.html#events_emitter_on_eventname_listener
+ *           @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+ * @param config Cypress configuration object to be updated
+ * @returns Updated configuration
+ */
+function setupPluginsAndConfig(
+    on: Cypress.Actions,
+    config: Cypress.ResolvedConfigOptions
+): Cypress.ResolvedConfigOptions {
```

Comment:
After applying the suggestion for typing `coverageTask` you will see an error when calling it, because `on` has a different type. You specified it as `Cypress.Actions`, and `Cypress.PluginConfig` expects `Cypress.PluginEvents`.

After taking a look at the type declarations for each of them and consulting [the list of plugin events in the docs](https://docs.cypress.io/api/plugins/writing-a-plugin.html#List-of-events), `Cypress.PluginEvents` is the correct type for `on` :slightly_smiling_face: 

However, we could do better :smile: Reusing types for the win here:

```suggestion
const setupPluginsAndConfig: Cypress.PluginConfig = (on, config) => {
```

A lot less code, and more type safety :slightly_smiling_face:

Reply:
I admit I didn't even realize it's possible to reuse a complete function definition in such a way - cool, thanks.

Reply:
No problem :smile: That's one of the nicest features of TS that I use quite often :slightly_smiling_face: